src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -123,13 +125,17 @@ public ObservableList<Person> asUnmodifiableObservableList() {
     /**
      * Returns the backing list as an unmodifiable {@code ObservableList} filtered by the given tag list.
      */
-    public ObservableList<Person> asUnmodifiableFilteredObservableList(Set<Tag> tagList) {
+    public ObservableList<Person> asUnmodifiableFilteredObservableList(Set<Tag> tagList) throws CommandException {
```

Comment:
This is a bad design as reading the list of person from addressbook shouldn't have anything to do with `Command`, you should do the checking in the `DownloadCommand.java`

Reply:
fixed